docs/HACKING.md

```diff
@@ -14,6 +14,14 @@ feature, please feel invited to fix it, and submit your work as a
 Please make sure to follow our [Coding Style](CODING_STYLE) when submitting
 patches. Also have a look at our [Contribution Guidelines](CONTRIBUTING).
 
+To start, run the following commands in the systemd git repository to set up
+git correctly:
+
+```shell
+$ git config submodule.recurse true
+$ git config fetch.recurseSubmodules on-demand
+```
+
```

Comment:
this can be dropped now, no?

Reply:
I think it's better to document explicitly, not everyone runs scripts from repos they clone

Reply:
meson hooks this in automatically... so if you run meson you are already set.

At the very least this is something to mention: that the meson build stuff will enable this for you

Reply:
@poettering Yes but it's perfectly possible to hack on systemd these days without ever running `meson` on the host at all. And changes to sources from `meson` running in mkosi are thrown away at the end of the build. So it's not guaranteed that these settings will always be applied so documenting them as well seems like a good idea.

Reply:
But then what @poettering said, it should be mentioned that running those commands is not necessary if meson is used. Maybe, "If you're not using meson at all, …".